src/database/schema.rs

```diff
@@ -43,6 +47,7 @@ CREATE TABLE vaults (
     derivation_index INTEGER NOT NULL,
     amount INTEGER NOT NULL,
     delegated INTEGER NOT NULL CHECK (delegated IN (0,1)),
+    should_cancel INTEGER CHECK (delegated IN (NULL, 0,1)),
```

Comment:
Should it be:
`should_cancel INTEGER CHECK (should_cancel IN (NULL,0,1)),`?

Reply:
Ouch, nice catch